components/esp_lvgl_port/src/lvgl8/esp_lvgl_port_knob.c

```diff
@@ -125,11 +131,17 @@ static void lvgl_port_encoder_read(lv_indev_drv_t *indev_drv, lv_indev_data_t *d
     assert(ctx);
 
     int32_t invd = iot_knob_get_count_value(ctx->knob_handle);
-    knob_event_t event = iot_knob_get_event(ctx->knob_handle);
+    knob_event_t event = ctx->event;
 
     if (last_v ^ invd) {
+
+        int32_t diff = (int32_t)((uint32_t)invd - (uint32_t)last_v);
+
+        diff += (event == KNOB_RIGHT && invd < last_v) ? CONFIG_KNOB_HIGH_LIMIT :
```

Comment:
Why we need this? 
`event` is only last event and when knob rotate to right by some steps then to left, the `event` will left, but more steps can be still right.

Reply:
Yeah, you're right. In some case, as follows:
Starting at 10, rotating right by 10 [20] (not captured by LVGL this time), and then rotating left by 5 [15].

This would result in a `rotate left` event, but with a positive diff, leading to an abnormal diff for the left rotation.

Should we ignore the diff?

Reply:
In your example case, there is still direction (diff as you naimed). I think, the direction is only in sign minus or plus value. That's all.

Reply:
Use interrupts to record the current diff value in real time. 
Accumulate the value when rotating in the same direction; reset to zero and start recalculating when the direction reverses.

Reply:
Why do you reset value when changed direction? I thought, it is enough to add or sub from latest value.

Reply:
If the encoder is rotated `10 steps to the right`, but `lvgl_port_encoder_read `does not capture it (assuming the UI is experiencing severe lag at this moment), and then it is rotated `5 steps to the left`, I think a result of `5 steps to the left` is `better` than `5 steps to the right`. 
Therefore, the count for the different direction should be reset.